### PR TITLE
Set Input Type for TensorRT convert tests

### PR DESCRIPTION
**Pull request by eee4017**
### PR types
Others

### PR changes
Others

### Description
We preprocess the test_trt_convert test to standardize the input type to either FP32 or FP16, ensuring a uniform input type.

### CODE REVIEW

File: test/ir/inference/auto_scan_test.py [hunk 1]
```diff
@@ -35,9 +34,12 @@
 import paddle
 import paddle.inference as paddle_infer
 from paddle.fluid.core import PassVersionChecker
+from paddle.static.log_helper import get_logger
 
 LOGLEVEL = os.environ.get("PADDLE_TEST_LOGLEVEL", "INFO").upper()
-logging.basicConfig(level=LOGLEVEL, format="%(message)s")
+logging = get_logger(
+    __name__, LOGLEVEL, fmt='%(asctime)s-%(levelname)s: %(message)s'
+)
```
Comment (eee4017) [inline test/ir/inference/auto_scan_test.py:42]: We changed the logging system to the paddle logger.

File: test/ir/inference/auto_scan_test.py [hunk 2]
```diff
@@ -747,10 +746,13 @@ def random_to_skip():
             if not skip_baseline:
                 # baseline: gpu run, we only test float32
                 gpu_config = self.create_inference_config(use_trt=False)
+                prog_config = prog_config.set_input_type(
```
Comment (eee4017) [inline test/ir/inference/auto_scan_test.py:749]: We reduce the input of the baseline to a smaller number of bits, which could potentially narrow the gap between the fp16 and fp32 computation.

File: test/ir/inference/program_config.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def __init__(
     def __repr__(self):
         return str({'shape': self.shape, 'lod': self.lod, 'dtype': self.dtype})
 
-    def astype(self, type: np.dtype):
+    def convert_type_inplace(self, type: np.dtype):
```
Comment (eee4017) [inline test/ir/inference/program_config.py:70]: The name has been altered to enhance readability.

File: test/ir/inference/test_trt_convert_activation.py [hunk 4]
```diff
@@ -178,21 +178,25 @@ def generate_trt_nodes_num(attrs, dynamic_shape):
         # for static_shape
         clear_dynamic_shape()
         self.trt_param.precision = paddle_infer.PrecisionType.Float32
+        program_config.set_input_type(np.float32)
```
Comment (eee4017) [inline test/ir/inference/test_trt_convert_activation.py:181]: For all files with the test_trt_convert test, we cast the input type of the input data in the program configuration before creating the sample configuration.

**Comment by eee4017** [comment]
Waiting for Dianhai or XiaoguangHu01 approval

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Comment by onecatcn** [comment]
@eee4017 please rerun PR-CI-APPROVAL

**Review by yuanlehome — APPROVED** [review]
LGTM

**Comment by yuanlehome** [comment]
我理解之后trt单测精度比较的容忍度更高了，之后很多单测的atol/rtol的设置是否需要批量改一下

**Comment by eee4017** [comment]
I think it could be merge

**Comment by eee4017** [comment]
I think it could be merged now.

**Pull request merged**